zipline/algorithm.py

```diff
@@ -510,15 +510,14 @@ def run(self, source, overwrite_sim_params=True,
                 self.sim_params.data_frequency,
             )
 
-        with ZiplineAPI(self):
-            # loop through simulated_trading, each iteration returns a
-            # perf dictionary
-            perfs = []
-            for perf in self.gen:
-                perfs.append(perf)
-
-            # convert perf dict to pandas dataframe
-            daily_stats = self._create_daily_stats(perfs)
+        # loop through simulated_trading, each iteration returns a
+        # perf dictionary
+        perfs = []
+        for perf in self.gen:
+            perfs.append(perf)
```

Comment:
@richafrank better yet, just pass the generator straight to _create_daily_stats.